lib/compiler/test.js

```diff
@@ -1,18 +1,52 @@
-import { createRequire } from 'module';
-import { pathToFileURL } from 'url';
-
-const require = createRequire(import.meta.url);
+const sleep = (ms) => new Promise((res) => setTimeout(res, ms));
 
 (async () => {
-  const { transform } = await import('@astrojs/compiler');
-  const template = await transform(`---
+  const { transform } = await import("@astrojs/compiler");
+  let i = 0;
+  try {
+    const value = await transform(
+      `---
 let value = 'world';
 ---
-<h1>Hello {value.split('').reverse().join('')}!</h1>`, { sourcemap: true });
+
+<style lang="scss" define:vars={{ a: 0 }}>
+div {
+  color: red;
+}
+</style>
+
+<div>Hello world!</div>
+
+<div>Ahhh</div>
+
+<style lang="scss">
+div {
+  color: green;
+}
+</style>
+`,
+      {
+        sourcemap: true,
+        // HOLY CRAP THIS ACTUALLY WORKS!
+        preprocessStyle: async (value, attrs) => {
+          let x = i++;
+          if (!attrs.lang) {
+            return null;
+          }
+          console.log(`Starting to preprocess style ${x} as ${attrs.lang}`);
+          await sleep(3000);
+          console.log(`Finished preprocessing ${x}`);
+          return value.replace('color', 'background');
```

Comment:
Should be very straightforward.

`preprocessStyle` is an async function that gets the contents of the `style` element and any compile-time known attrs (we only care about `lang` now, but maybe others in the future!)

Simulating some expensive work with `sleep` here.